corehq/apps/es/README.rst

```diff
@@ -159,7 +159,20 @@ Reindex Procedure Details
 
        ./manage.py elastic_sync_multiplexed ElasticBook
 
-3. Disable multiplexing for the index.
+3. Perform a primary/secondary "swap" operation one or more times as desired to
+   run a "live test" on the new (secondary) index while keeping the old
+   (primary) index up-to-date.
+
+   - Reconfigure the adapter by swapping the "primary" and "secondary" index
```

Comment:
Same question about having this be environment specific. Depending on the size of these indices, these processes will take way different amounts of times to finish and rely on third party hosters with unknown deploy schedules having run specific management commands already. Is there a way to make this safer for them, so that these potentially dangerous commands don't happen automatically in environments we don't control?

Reply:
If we need to add "rapid" index swapping (primary/secondary), this design doesn't prohibit it (it can be added later as needed). For the sake of this design I'm viewing it as an optimization, and the initial implementation is designed to handle reindexing via migrations alone.

> Is there a way to make this safer for them, so that these potentially dangerous commands ...

The safety is guaranteed by the migrations framework. Updating and running the management commands will not be dangerous because such operations will expect a specific "existing" state. That is, the migration (and "sync" management command) will fail if the Elasticsearch index state is not what it expects.